src/compas_fab/ghpython/components/Cf_ConstraintsFromConfiguration/code.py

```diff
@@ -0,0 +1,24 @@
+"""
+Create a position and an orientation constraint from a robot configuration.
+
+COMPAS FAB v0.25.0
+"""
+import math
+
+from ghpythonlib.componentbase import executingcomponent as component
+
+
+class ConstraintsFromTargetConfiguration(component):
+    def RunScript(self, robot, target_configuration, tolerance_above, tolerance_below, group_name):
+        if robot and target_configuration:
+            tolerance_above = tolerance_above or [math.radians(1)] * 6
+            tolerance_below = tolerance_below or [math.radians(1)] * 6
```

Comment:
This one is a bit more involved to be generic: it won't always be 6 values and it won't always be in radians. So, you need to get a `get_configurable_joints(group)` for the robot, and add defaults based on the type of each joint (for PRIMASTIC and PLANAR, the unit is meters, for all the rest, radians).

As an example of a more annoying case, the SCARA robots have 4 axes (revolute, revolute, prismatic, revolute), so, it should end up in `radians(1), radians(1), 0.001, radians(1)` (the prismatic could default to 1mm, but it is defined in meters).

Reply:
Good point. 
Is there maybe a more generic attribute I should check? (e.g. `is_scalable()`) or is this not directly related to the tolerance unit.

Reply:
Yes, i think you can rely on `is_scalable` because angles cannot be scaled so this has matching semantics.